### PR TITLE
Add Synonyms API notebook

### PR DESCRIPTION
**Pull request by Mikep86**
Adds a synonyms API notebook that shows how to use the new synonyms API using Colab

### CODE REVIEW

**Review by kderusso — COMMENTED** [review]
LGTM, I'd love to get a review from the @elastic/search-experiences-team too!

File: notebooks/search/06-synonymys-api.ipynb
Comment (carlosdelest) [inline notebooks/search/06-synonymys-api.ipynb:283]:
```suggestion
        "A synonym token graph allows to use multi-word synonyms. Using it in a search analyzer allow us to  update synonyms without reindexing.\n",
```

File: notebooks/search/06-synonymys-api.ipynb
Comment (carlosdelest) [reply notebooks/search/06-synonymys-api.ipynb:285]: IMO I don't think we should - let's keep interlinking to Elastic related content

File: notebooks/search/06-synonymys-api.ipynb
Comment (carlosdelest) [inline notebooks/search/06-synonymys-api.ipynb:446]:
```suggestion
        "We didn't get any results! There are some books that use the terms \"artificial intelligence\", but not \"AI\". Let's try using the Synonyms API to add a new synonym rule for AI so the previous query returns results.\n",
```

File: notebooks/search/06-synonymys-api.ipynb
Comment (carlosdelest) [inline notebooks/search/06-synonymys-api.ipynb:515]: Should we add some conclusion like sentences like "Using synonyms API allows you to..."? I'm including search experiences team to review as well as @kderusso mentioned

**Review by carlosdelest — APPROVED** [review]
LGTM! Some minor comments. I'd wait for search-experiences review before merging.

Thanks for creating this! 💯

**Comment by miguelgrinberg** [comment]
Hey @Mikep86! I tested your notebook and everything works. I want to suggest that for the code cells that return search results you leave the output stored in the notebook. You can look at the "00-quick-start" notebook for an example.

The reason is that we (search experiences) are working on an testing script that will run through our notebooks as new versions of ES are released. The intention is that for every code cell that is stored with its output the test would pass only if the output generated during the test run matches (minus unimportant differences such as randomly geneated ids).

**Review by miguelgrinberg — APPROVED** [review]

File: notebooks/search/06-synonymys-api.ipynb
Comment (Mikep86) [reply notebooks/search/06-synonymys-api.ipynb:285]: Ok, I can link to [The same, but different: Boosting the power of Elasticsearch with synonyms](https://www.elastic.co/blog/boosting-the-power-of-elasticsearch-with-synonyms), but it doesn't go into nearly as much detail about token streams & multi-word synonyms. It would be nice if we had an internal doc/blog that covered that in more detail.

**Review by saarikabhasi — APPROVED** [review]
Tested the notebook. Code changes LGTM ! 

It's worth adding this new file to [Readme.md](https://github.com/elastic/elasticsearch-labs/blob/main/README.md#search) file under the section `search` for easy access from readme.

File: notebooks/search/06-synonymys-api.ipynb
Comment (Mikep86) [reply notebooks/search/06-synonymys-api.ipynb:280]: Added context to the paragraph just below the list

**Review by kderusso — APPROVED** [review]
Looks great! I think we still want the responses in here But otherwise 💯 🚀 🚢

**Comment by Mikep86** [comment]
@saarikabhasi I updated the readme under `search` and at the root of the repo, but it looks a little odd right now because the entry for the query rules notebook in the `search` README is missing.

Should we address that here or on in a separate PR?

cc: @kderusso

**Comment by kderusso** [comment]
@Mikep86 Thanks for the catch! If you're OK adding it in this PR I'd be much appreciative. 🙇

File: notebooks/search/README.md
Comment (Mikep86) [inline notebooks/search/README.md:76]: @kderusso Look good?

Comment (kderusso) [inline notebooks/search/README.md:76]:
```suggestion
- Use the query rules management APIs to create and edit promotional rules based on contextual queries
- Apply these query rules by using the `rule_query` in Query DSL
```

File: notebooks/search/README.md
Comment (kderusso) [inline notebooks/search/README.md:82]:
```suggestion
- Use the synonyms management API to create a synonyms set to enhance your search recall
```

**Review by kderusso — APPROVED** [review]
LGTM!

**Pull request merged**
